### PR TITLE
adds documentation for flag --primary-key-type=uuid passed to Rails model generator

### PR DESCRIPTION
**Pull request by jasonfb**
please see https://github.com/rails/rails/issues/46229

### CODE REVIEW

File: guides/source/active_record_postgresql.md [hunk 1]
```diff
@@ -504,6 +504,16 @@ irb> device.id
...
+To use the Rails model generator to make a table with UUID as the primary key, pass `--primary-key-type=uuid` as a flag to the model generator.
+
+When building a foreign key that will reference this a UUID, treat `uuid` as the field type, for example:
+
+(in this arbitrary example, a 'case' belongs to a device and so has a foriegn key `device_id` to the `devices` table we made above)
+
+`rails generate model Case device_id:uuid`
+
+
+
```
Comment (jasonfb) [inline guides/source/active_record_postgresql.md:516]:
```suggestion
```

File: guides/source/active_record_postgresql.md [hunk 2]
```diff
@@ -504,6 +504,11 @@ irb> device.id
 NOTE: `gen_random_uuid()` (from `pgcrypto`) is assumed if no `:default` option was
 passed to `create_table`.
 
+To use the Rails model generator to make a table with UUID as the primary key, pass `--primary-key-type=uuid` as a flag to the model generator.
+
+When building a foreign key that will reference this a UUID, treat `uuid` as the field type, for example:
```
Comment (jasonfb) [inline guides/source/active_record_postgresql.md:509]:
```suggestion
When building a foreign key that will reference this UUID, treat `uuid` as the native field type, for example:
```

File: guides/source/active_record_postgresql.md [hunk 3]
```diff
@@ -504,6 +504,11 @@ irb> device.id
 NOTE: `gen_random_uuid()` (from `pgcrypto`) is assumed if no `:default` option was
 passed to `create_table`.
 
+To use the Rails model generator to make a table with UUID as the primary key, pass `--primary-key-type=uuid` as a flag to the model generator.
```
Comment (eileencodes) [inline guides/source/active_record_postgresql.md:507]:
```suggestion
To use the Rails model generator for a table using UUID as the primary key, pass `--primary-key-type=uuid` to the model generator.
```

File: guides/source/active_record_postgresql.md [hunk 4]
```diff
@@ -504,6 +504,11 @@ irb> device.id
...
+
+When building a foreign key that will reference this UUID, treat `uuid` as the native field type, for example:
```
Comment (eileencodes) [inline guides/source/active_record_postgresql.md:509]:
```suggestion
When building a model with a foreign key that will reference this UUID, treat `uuid` as the native field type, for example:
```

File: guides/source/active_record_postgresql.md [hunk 5]
```diff
@@ -504,6 +504,11 @@ irb> device.id
...
+
+`rails generate model Case device_id:uuid`
```
Comment (eileencodes) [inline guides/source/active_record_postgresql.md:511]: This needs a new line but also needs to be code-blocked with ruby like the above examples.

```suggestion
`rails generate model Case device_id:uuid`

```

**Review by eileencodes — CHANGES_REQUESTED** [review]
Hey @jasonfb thanks for the PR. This needs some changes before it's mergable.

1) Docs should be wrapped to 80 characters
2) New line code examples should be wrapped in a "```ruby" block, see above code for an example.
3) I've added some wording changes to make the language clearer.
4) When you're done squash your commits into 1 commit, we don't merge PRs with multiple commits unless they tell a specific and clear story.

Thanks!

**Comment by jasonfb** [comment]
ok thanks @eileencodes for the feedback. I squashed it down. Not sure what you mean by "Docs should be wrapped to 80 characters"? I think all the lines I added are under 80 characters long. thanks.

File: guides/source/active_record_postgresql.md [hunk 6]
```diff
@@ -504,6 +504,21 @@ irb> device.id
 NOTE: `gen_random_uuid()` (from `pgcrypto`) is assumed if no `:default` option was
 passed to `create_table`.
 
+To use the Rails model generator for a table using UUID as the primary key, pass `--primary-key-type=uuid` to the model generator.
```
Comment (eileencodes) [inline guides/source/active_record_postgresql.md:507]:
```suggestion
To use the Rails model generator for a table using UUID as the primary key, pass
`--primary-key-type=uuid` to the model generator.
```

File: guides/source/active_record_postgresql.md [hunk 7]
```diff
@@ -504,6 +504,21 @@ irb> device.id
...
+
+For example:
+
+```ruby
+rails generate model Device --primary-key-type=uuid kind:string
+```
+
+When building a model with a foreign key that will reference this UUID, treat `uuid` as the native field type, for example:
```
Comment (eileencodes) [inline guides/source/active_record_postgresql.md:515]:
```suggestion
When building a model with a foreign key that will reference this UUID, treat `uuid`
as the native field type, for example:
```

**Review by eileencodes — CHANGES_REQUESTED** [review]
Wrapped to 80 characters means that lines shouldn't exceed 80 characters in length. I've left a code suggestion so you can see where the breaks should be 👍🏼

**Comment by jasonfb** [comment]
oh, I got it. thank you. this is hard-wrapped to 80 chars now.

**Comment by eileencodes** [comment]
Thank you @jasonfb!

**Comment by jasonfb** [comment]
@eileencodes - you know I discovered a bug that makes this wrong — it appears to create a table that **_does_** have a ` id: :uuid ` set in the create_table definition, but then the generated migration doesn't actually contain the fields specified. 

so I guess the bug is that you can't use the flag and also pass fields to the model generator?

**Comment by skipkayhil** [comment]
> so I guess the bug is that you can't use the flag and also pass fields to the model generator?

If you have reproduction steps, can you open a new issue? Replying to old PRs usually won't get as much visibility

**Pull request merged**